x-pack/metricbeat/module/oracle/connection.go

```diff
@@ -0,0 +1,84 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package oracle
+
+import (
+	"database/sql"
+
+	"gopkg.in/goracle.v2"
+
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+
+	// Register driver.
+	_ "gopkg.in/goracle.v2"
+
+	"github.com/pkg/errors"
+)
+
+// ConnectionDetails contains all possible data that can be used to create a connection with
+// an Oracle db
+type ConnectionDetails struct {
+	Username string   `config:"username"    validate:"nonzero"`
+	Password string   `config:"password"    validate:"nonzero"`
+	Hosts    []string `config:"hosts"    validate:"nonzero"`
```

Comment:
Umm, `hosts` is required, but `username` and `password` are not, should they have different validate options?

Reply:
Thank you for this! Only hosts is mandatory now that we support connection strings.